tests/test_cli_event_source_guards.py

```
@@ -0,0 +1,79 @@
+}
+
+
+def validate_incremental_stream_scan(source: str) -> None:
+    start = source.index("    private func readStreamLine(")
+    end = source.index("    private func waitForReadableStream(", start)
+    body = source[start:end]
+
+    declaration = "private var streamLineSearchOffset = 0"
+    incremental_search = "streamReadBuffer[searchStart...].firstIndex(of: 0x0A)"
+    if declaration not in source:
+        raise AssertionError("CLI event reader is missing its persistent scan offset")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)